braindecode/training/scoring.py

```diff
@@ -295,3 +297,59 @@ def on_epoch_end(self, net, dataset_train, dataset_valid, **kwargs):
                 cached_net, dataset_train, self.y_trues_
             )
         self._record_score(net.history, current_score)
+
+
+def predict_trials(module, dataset, return_targets=True):
+    """Create trialwise predictions and optionally also return trialwise
+    labels from cropped dataset given module.
+
+    Parameters
+    ----------
+    module: torch.nn.Module
+        A pytorch model implementing forward.
+    dataset: braindecode.datasets.BaseConcatDataset
+        A braindecode dataset to be predicted.
+    return_targets: bool
+        If True, additionally returns the trial targets.
+
+    Returns
+    -------
+        trial_predictions: np.ndarray
+            3-dimensional array (n_trials x n_classes x n_predictions), where
+            the number of predictions depend on the chosen window size and the
+            receptive field of the network.
+        trial_labels: np.ndarray
+            2-dimensional array (n_trials x n_targets) where the number of
+            targets depends on the decoding paradigm and can be either a single
+            value, multiple values, or a sequence.
+    """
+    # we have a cropped dataset if there exists at least one trial with more
+    # than one compute window
+    cropped_data = sum(dataset.get_metadata()['i_window_in_trial'] != 0) > 0
+    if not cropped_data:
+        raise ValueError('This function was designed to predict trials from '
+                         'cropped datasets. This is a trialwise dataset.')
```

Comment:
```suggestion
    more_than_one_window = sum(dataset.get_metadata()['i_window_in_trial'] != 0) > 0
    if not more_than_one_window:
        warnings.warn('This function was designed to predict trials from '
            'cropped datasets, which typically have multiple compute windows per trial .'
            'The given dataset has exactly one window per trial,')
```
Is it sure that this must be the case? I think this is not necessarily true. Maybe at least downgrade to a warning instead of an error?

Reply:
I could not come up with a case where it does not hold. Isn't it the definition of trialwise decoding to have one window per trial?

Reply:
one can consider (and at least that's what I discussed with Tonio as well as far as I recall) trialwise decoding implies single window and *single prediction* per trial, whereas if you have single window but multiple predictions, than you can still consider it cropped decoding. and all the code should also run fine

Reply:
Ok. Then I will update according to your suggestions